test/e2e/services.go

```diff
@@ -95,14 +96,33 @@ func startServicesForLogs(t *testing.T, e e2e.Environment) (
 }
 
 func startServicesForTraces(t *testing.T, e e2e.Environment) (otlpGRPCEndpoint, jaegerExternalHttpEndpoint, jaegerInternalHttpEndpoint string) {
+	prometheus := e.Runnable("prometheus").
+		WithPorts(
+			map[string]int{
+				"http": 9090,
+			}).
+		Init(e2e.StartOptions{Image: prometheusImage})
+	testutil.Ok(t, e2e.StartAndWaitReady(prometheus))
+
+	// The endpoint points to localhost
+	// parse the port and use it with host IP address later
+	promHttpEndpoint := prometheus.Endpoint("http")
+	promHostPort := promHttpEndpoint[strings.Index(promHttpEndpoint, ":")+1:]
```

Comment:
We can probably replace all this with `e2edb.NewPrometheus()` and you can use the port later with `.InternalEndpoint("http")`

Reply:
+1 changed